dialer.go

```diff
@@ -50,97 +62,124 @@ func NewDialer(opts *DialerOpts) Dialer {
 	if opts.WindowSize <= 0 {
 		opts.WindowSize = defaultWindowSize
 	}
+	if opts.MaxLiveConns <= 0 {
+		opts.MaxLiveConns = 1
+	}
 	if opts.MaxStreamsPerConn <= 0 || opts.MaxStreamsPerConn > maxID {
 		opts.MaxStreamsPerConn = maxID
 	}
-	log.Debugf("Initializing Dialer with   windowSize: %v   maxPadding: %v   maxStreamsPerConn: %v   pingInterval: %v   cipher: %v",
+
+	if opts.RedialSessionInterval <= 0 {
+		opts.RedialSessionInterval = 5 * time.Second
+	}
+	log.Debugf("Initializing Dialer with   windowSize: %v   maxPadding: %v   maxLiveConns: %v  maxStreamsPerConn: %v   pingInterval: %v   cipher: %v",
 		opts.WindowSize,
 		opts.MaxPadding,
+		opts.MaxLiveConns,
 		opts.MaxStreamsPerConn,
 		opts.PingInterval,
 		opts.Cipher)
+	liveSessions := make(chan sessionInf, opts.MaxLiveConns)
+	liveSessions <- nullSession{}
```

Comment:
What's the purpose of the nullSession?

Reply:
Without one session in the channel, https://github.com/getlantern/lampshade/pull/30/files#diff-6e2887d95404d95d21e5ef26c7d9c500R159 would block when it's called for the first time, but dialing a session when dialer is created is a waste of resource if it's never used.

Reply:
Oh you reminded me that `EMARTT` may be called on a `nullSession` and panic. Will fix.